### DISCUSSION TITLE
Unable to have a list of lists

### DISCUSSION THREAD

**Discussion by geyang** [prisma/prisma#13638] [Q&A]
What is the best way to write a schema for the following data structure? this is mandated by the data we are trying to store and have no control over.

```
example_data = {
     simple_key: 10,
     an_array: [
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
    ]
}
```

Would you kindly provide an example schema for how to accomplish? Again, it is a nested list item.

**Comment by austincrim** [comment]
Hey @geyang 👋,

It is not possible to have lists of lists in the Prisma schema. You will need to use a `Json` type field to workaround this.

We have an [outstanding issue](https://github.com/prisma/prisma/issues/3219) to specify types for JSON fields that you could +1.

**Reply by geyang** [reply]
JSON type being a list in this case?

**Reply by austincrim** [reply]
Yeah, so if you go with a JSON type in order to have a list of lists, you will lose strong typing for now. [This issue](https://github.com/prisma/prisma/issues/3219) will add strong typings to JSON so you can have the full Prisma experience.

**Reply by geyang** [reply]
I tried JSON typing it does not work. It says only object JSON types are supported, not list types

**Reply by austincrim** [reply]
You would need to include this whole object in the JSON field:
```
{
     simple_key: 10,
     an_array: [
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
        [{a_key: 10, b_key: 42}, {end_of_it: 69}],
    ]
}
```

It needs to be an object at the top level.

**Reply by geyang** [reply]
You mean using a JSON type for the entire collection? How do I do that?

**Reply by austincrim** [reply]
If you have a `JSON` type field in your schema, provide the whole object when writing data:

```ts
await prisma.user.create({
  data: {
    jsonStuff: {      // top level is an object
      simple_key: 10,
      an_array: [     // array of arrays here
        [{ a_key: 10, b_key: 42 }, { end_of_it: 69 }],
        [{ a_key: 10, b_key: 42 }, { end_of_it: 69 }],
        [{ a_key: 10, b_key: 42 }, { end_of_it: 69 }],
        [{ a_key: 10, b_key: 42 }, { end_of_it: 69 }],
        [{ a_key: 10, b_key: 42 }, { end_of_it: 69 }]
      ]
    }
  }
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
